server/lib/activity-event.js

```diff
@@ -0,0 +1,54 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+var _ = require('lodash');
+var Promise = require('bluebird');
+
+var ALLOWED_QUERY_PARAMS = [
+  'context',
+  'entrypoint',
+  'migration',
+  'service',
+  'utm_campaign',
+  'utm_content',
+  'utm_medium',
+  'utm_source',
+  'utm_term'
+];
+var MAX_PARAM_LENGTH = 100;
+
+module.exports = function (event, data, request) {
+  var queryParams = _.pick(request.query, ALLOWED_QUERY_PARAMS);
+  var eventData = _.assign({
```

Comment:
Not a comment on whether this is correct or not, I didn't know about _.assign before yesterday. What is the advantage of _.assign over _.extend?